web-search/web-search/src/durability.rs

```diff
@@ -0,0 +1,289 @@
+use crate::exports::golem::web_search::web_search::{Guest, SearchError, SearchParams};
+use std::marker::PhantomData;
+
+/// Wraps a web search implementation with custom durability
+pub struct DurableWebSearch<Impl> {
+    phantom: PhantomData<Impl>,
+}
+
+/// Trait to be implemented in addition to the web search `Guest` trait when wrapping it with `DurableWebSearch`.
+pub trait ExtendedGuest: Guest + 'static {
+    /// Creates an instance of the web search specific `SearchSession` without wrapping it in a `Resource`
+    fn unwrapped_search_session(params: SearchParams) -> Result<Self::SearchSession, SearchError>;
+
+    /// Creates a search session from stored state (for recovery)
+    fn session_from_state(
```

Comment:
I think this name / description are not great. session_for_page would be a better fit imo